src/supla/sensor/HX711.h

```diff
@@ -0,0 +1,161 @@
+      value = (value < 0 && value > -1) ? 0 : value;
+    }
+    return value;
+  }
+
+  void onInit() {
+    hx711Sensor.begin();
+    hx711Sensor.setTareOffset(tareOffset);
+    uint64_t stabilizingTime = 2000;
+    bool _tare = false;
+    hx711Sensor.start(stabilizingTime, _tare);
+    if (hx711Sensor.getTareTimeoutFlag()) {
+      SUPLA_LOG_DEBUG("HX711 timeout, check connection");
+    } else {
+      calFactor = (calValue > 0) ? calValue : defCalValue;
+      hx711Sensor.setCalFactor(calFactor);
+      SUPLA_LOG_DEBUG("HX711 startup is complete");
+    }
+    while (!hx711Sensor.update()) {}
+    SUPLA_LOG_INFO("HX711 calibration value: %.2f", hx711Sensor.getCalFactor());
+    SUPLA_LOG_INFO(
+        "HX711 measured conversion time ms: %.2f",
+         hx711Sensor.getConversionTime());
+    SUPLA_LOG_DEBUG(
+        "HX711 measured sampling rate HZ: %.2f", hx711Sensor.getSPS());
+    SUPLA_LOG_INFO(
+        "HX711 measured settlingtime ms: %d", hx711Sensor.getSettlingTime());
+    if ((hx711Sensor.getSPS() < 7) || (hx711Sensor.getSPS() > 100)) {
+      SUPLA_LOG_DEBUG("HX711 sampling rate is abnormal, check connection");
+    }
+    channel.setNewValue(getValue());
+  }
+
+  void calibrateScales(float knownMass = 0) {
+    if (!readyToCalibration) {
+      SUPLA_LOG_DEBUG("HX711 please do tare");
+    } else {
+      hx711Sensor.update();
+      hx711Sensor.refreshDataSet();
+      if (knownMass > 0) {
+        calValue = hx711Sensor.getNewCalibration(knownMass);
+        SUPLA_LOG_DEBUG("HX711 new calibration value: %.2f", calValue);
+        Supla::Storage::ScheduleSave(100);
+      } else {
+        SUPLA_LOG_DEBUG("HX711 known mass must be greater than zero");
+      }
+    }
+    readyToCalibration = false;
+  }
+
+  void tareScales() {
+    hx711Sensor.tare();
+    tareOffset = hx711Sensor.getTareOffset();
+    SUPLA_LOG_DEBUG("HX711 new tare offset: %d", tareOffset);
+    Supla::Storage::ScheduleSave(100);
+    readyToCalibration = true;
+  }
+
+  void iterateAlways() {
+    if (hx711Sensor.update()) {
+      value = hx711Sensor.getData();
+    }
+    if (millis() - lastReadTime > 500) {
+      lastReadTime = millis();
+      channel.setNewValue(getValue());
+    }
+  }
+
+  void onLoadState() {
+    HX711ConfigData data = HX711ConfigData();
+    if (Supla::Storage::ReadState((unsigned char *)&data, sizeof(data))) {
+      tareOffset = data.tareOffset;
+      calValue = data.calValue;
+    }
+     SUPLA_LOG_DEBUG(
+        "HX711 settings restored from storage. Tare offset: %d; "
+        "calibration value: %.2f", tareOffset, calValue);
+  }
+
+  void onSaveState() {
+    HX711ConfigData data;
+    data.tareOffset = tareOffset;
+    data.calValue = calValue;
+
+    Supla::Storage::WriteState((unsigned char *)&data, sizeof(data));
+  }
+
+  float getTareOffset() {
+    return tareOffset;
+  }
+
+  float getCalFactor() {
+    return calFactor;
+  }
+
+ protected:
+  ::HX711_ADC hx711Sensor;
+  int sdaPin;
+  int sclPin;
```

Comment:
those variables are not used. Remove it

Reply:
ok

Reply:
Without these variables, compilation in the arduino IDE fails

Reply:
Yes, you have to also remove them from initialization list of constructor

Reply:
Done